zerver/webhooks/gitlab/view.py

```diff
@@ -474,6 +476,97 @@ def get_deployment_event_body(payload: WildValue, include_title: bool) -> str:
     return deployment_event_body_map[deployment_status]
 
 
+def get_emoji_event_transformed_type(payload: WildValue, type: str) -> str:
+    if type == "MergeRequest":
+        return "MR"
+    elif type == "Note":
+        event_type = payload["note"]["noteable_type"].tame(check_string)
+        return get_emoji_event_transformed_type(payload, event_type)
+    elif type == "DesignManagement::Design":
+        return "design"
+    return type.lower()
+
+
+def get_emoji_event_subtype_message(type: str) -> str:
+    return "a comment" if type == "Note" else ""
+
+
+def get_emoji_event_url_id(payload: WildValue, type: str) -> tuple[str, str]:
+    if type == "design":
+        comment_url, _, design_name = parse_design_comment(
+            payload["note"], payload["project"]["web_url"].tame(check_string)
+        )
+        return comment_url, design_name
+
+    url = payload["object_attributes"]["awarded_on_url"].tame(check_string)
+
+    # Extract the last numeric ID in the URL path before any '#' fragment.
+    # Example:
+    # https://gitlab.com/abc/def/issues/123 → "123"
+    # https://gitlab.com/abc/def/-/merge_requests/456#note_789 → "456"
+    clean_url = url.split("#")[0]
+    match = re.search(r"/(\d+)(?:/)?$", clean_url)
+    assert match is not None
+    return url, match.group(1)
+
+
+def get_emoji_event_topic_title(
+    payload: WildValue, awardable_type: str, type: str, use_merge_request_title: bool
+) -> str:
+    """
+    The payload format depends on the event target type (MR / issue / snippet).
+    Get topic from the corresponding payload section's title.
+    """
+    event_type = ""
+    if type == "MR":
+        event_type = "merge_request" if use_merge_request_title else ""
+    elif type == "issue":
+        event_type = "issue" if awardable_type == "Note" else "work_item"
+    elif type == "snippet":
+        event_type = "project_snippet"
+
+    if event_type:
+        return payload[event_type]["title"].tame(check_string)
+
+    return ""
+
+
+def get_emoji_event_number_sign(type: str) -> str:
+    return "" if type == "design" else "#"
+
+
+def get_emoji_event_body(action: str, payload: WildValue, include_title: bool) -> str:
+    transformed_action = {"award": "added", "revoke": "removed"}.get(action, "reacted")
+    emoji = payload["object_attributes"]
+    awardable_type = emoji["awardable_type"].tame(check_string)
+    transformed_type = get_emoji_event_transformed_type(payload, awardable_type)
+    topic_title = get_emoji_event_topic_title(
```

Comment:
I don't see why we need the title at all. `include_title` looks sufficient to me.